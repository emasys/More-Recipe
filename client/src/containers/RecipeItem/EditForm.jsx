import React from 'react';
import Textarea from 'react-textarea-autosize';
import PropTypes from 'prop-types';

import { list as categoryList } from '../../components/CategoryList';

/**
 *
 *
 * @param {object} props
 *
 * @returns {JSX.Element}
 * react element
 */
const EditForm = props => (
  <div>
    <form onSubmit={props.handleSubmit}>
      <ul className="form row justify-content-center">
        <li className="col-lg-8 col-sm-12">
          <label>Recipe Title</label>
          <input
            type="text"
            required
            placeholder="Recipe Name"
            className="col-lg-11 col-sm-12"
            name="recipe"
            defaultValue={props.state.name}
          />
          <div className={`text-danger ${props.state.error}`} id="recipe_error">
            A recipe with this name already exist
          </div>
        </li>
        <li className="col-lg-8 col-sm-12">
          <label>
            Ingredients{' '}
            <small>
              <em>separate each ingredient with a comma {','}</em>
            </small>
          </label>
          <Textarea
            className="col-lg-11 col-sm-12"
            id="FormControlTextarea"
            name="ingredients"
            minRows={1}
            maxRows={50}
            defaultValue={props.state.ingredients}
          />
        </li>
        <li className="col-lg-8 col-sm-12">
          <label>Direction</label>
          <Textarea
            className="col-lg-11 col-sm-12"
            name="direction"
            minRows={1}
            maxRows={50}
            defaultValue={props.state.direction}
          />
        </li>

        <li className="col-lg-8 col-sm-12">
          <label>
            Description{' '}
            <small>
              <em>400 characters</em>
            </small>
          </label>
          <Textarea
            className="col-lg-11 col-sm-12"
            id="FormControlTextarea"
            defaultValue={props.state.description}
            name="description"
            minRows={1}
            maxRows={50}
            maxLength={400}
          />
        </li>
        <li className="col-lg-8 col-sm-12">
          <label>Category</label>
          <select
            name="category"
            className="col-12 "
            style={{ height: '50px' }}
            defaultValue={props.state.category}
          >
            {categoryList.map(item => (
              <option value={item} key={item} className="text-capitalize">
                {item}
              </option>
            ))}
          </select>
        </li>
        <li className=" col-12 ">
          <input
            type="submit"
            value="save"
            id="submit"
            className="btn bg-dark hovered"
          />
        </li>
      </ul>
    </form>
  </div>
);

EditForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  state: PropTypes.object.isRequired
};

export default EditForm;
