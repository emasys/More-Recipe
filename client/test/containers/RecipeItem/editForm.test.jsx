import React from 'react';
import { shallow } from 'enzyme';

// Component
import EditForm from '../../../src/containers/RecipeItem/EditForm';

let wrapper = null;

describe('Test suite for add recipe page', () => {
  const props = {
    handleSubmit: jest.fn(),
    goBack: jest.fn(),
    state: {}
  };

  beforeEach(() => {
    wrapper = shallow(<EditForm {...props} />);
  });
  it('Should render without errors', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('Should complete form and submit', () => {
    wrapper
      .find("input[name='recipe']")
      .simulate('change', { target: { value: 'How to boil water' } });
    wrapper
      .find("TextareaAutosize[name='description']")
      .simulate('change', { target: { value: 'A regular food' } });
    wrapper
      .find("TextareaAutosize[name='direction']")
      .simulate('change', { target: { value: 'Just do it' } });
    wrapper
      .find("TextareaAutosize[name='ingredients']")
      .simulate('change', { target: { value: 'water, salt' } });
    expect(wrapper.find("input[type='submit']").length).toEqual(1);
    wrapper.find('form').simulate('submit');
    wrapper.find('button.btn.btn-block.btn-lg.col-12.bg-warning.hovered').simulate('click');
    expect(props.goBack).toHaveBeenCalled();
    expect(props.handleSubmit).toBeCalled();
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
