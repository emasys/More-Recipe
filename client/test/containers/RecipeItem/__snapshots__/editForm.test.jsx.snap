// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for add recipe page should match snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <ul
      className="form row justify-content-center"
    >
      <li
        className="col-lg-8 col-sm-12"
      >
        <label>
          Recipe Title
        </label>
        <input
          className="col-lg-11 col-sm-12"
          name="recipe"
          placeholder="Recipe Name"
          required={true}
          type="text"
        />
        <div
          className="text-danger undefined"
          id="recipe_error"
        >
          A recipe with this name already exist
        </div>
      </li>
      <li
        className="col-lg-8 col-sm-12"
      >
        <label>
          Ingredients
           
          <small>
            <em>
              separate each ingredient with a comma 
              ,
            </em>
          </small>
        </label>
        <TextareaAutosize
          className="col-lg-11 col-sm-12"
          id="FormControlTextarea"
          maxRows={50}
          minRows={1}
          name="ingredients"
          onChange={[Function]}
          onHeightChange={[Function]}
          useCacheForDOMMeasurements={false}
        />
      </li>
      <li
        className="col-lg-8 col-sm-12"
      >
        <label>
          Direction
        </label>
        <TextareaAutosize
          className="col-lg-11 col-sm-12"
          maxRows={50}
          minRows={1}
          name="direction"
          onChange={[Function]}
          onHeightChange={[Function]}
          useCacheForDOMMeasurements={false}
        />
      </li>
      <li
        className="col-lg-8 col-sm-12"
      >
        <label>
          Description
           
          <small>
            <em>
              400 characters
            </em>
          </small>
        </label>
        <TextareaAutosize
          className="col-lg-11 col-sm-12"
          id="FormControlTextarea"
          maxLength={400}
          maxRows={50}
          minRows={1}
          name="description"
          onChange={[Function]}
          onHeightChange={[Function]}
          useCacheForDOMMeasurements={false}
        />
      </li>
      <li
        className="col-lg-8 col-sm-12"
      >
        <label>
          Category
        </label>
        <select
          className="col-12 "
          id="selectBox"
          name="category"
          style={
            Object {
              "height": "50px",
            }
          }
        >
          <option
            className="text-capitalize"
            key="Breakfast"
            value="Breakfast"
          >
            Breakfast
          </option>
          <option
            className="text-capitalize"
            key="Brunch"
            value="Brunch"
          >
            Brunch
          </option>
          <option
            className="text-capitalize"
            key="Lunch"
            value="Lunch"
          >
            Lunch
          </option>
          <option
            className="text-capitalize"
            key="Snacks"
            value="Snacks"
          >
            Snacks
          </option>
          <option
            className="text-capitalize"
            key="Appetisers"
            value="Appetisers"
          >
            Appetisers
          </option>
          <option
            className="text-capitalize"
            key="Dinner"
            value="Dinner"
          >
            Dinner
          </option>
          <option
            className="text-capitalize"
            key="Soups"
            value="Soups"
          >
            Soups
          </option>
          <option
            className="text-capitalize"
            key="Noodles"
            value="Noodles"
          >
            Noodles
          </option>
          <option
            className="text-capitalize"
            key="Rice"
            value="Rice"
          >
            Rice
          </option>
          <option
            className="text-capitalize"
            key="Pasta"
            value="Pasta"
          >
            Pasta
          </option>
          <option
            className="text-capitalize"
            key="Meat"
            value="Meat"
          >
            Meat
          </option>
          <option
            className="text-capitalize"
            key="Poultry"
            value="Poultry"
          >
            Poultry
          </option>
          <option
            className="text-capitalize"
            key="Seafood"
            value="Seafood"
          >
            Seafood
          </option>
          <option
            className="text-capitalize"
            key="Vegetarian"
            value="Vegetarian"
          >
            Vegetarian
          </option>
          <option
            className="text-capitalize"
            key="Sides"
            value="Sides"
          >
            Sides
          </option>
          <option
            className="text-capitalize"
            key="Sauces"
            value="Sauces"
          >
            Sauces
          </option>
          <option
            className="text-capitalize"
            key="Baking"
            value="Baking"
          >
            Baking
          </option>
          <option
            className="text-capitalize"
            key="Desserts"
            value="Desserts"
          >
            Desserts
          </option>
          <option
            className="text-capitalize"
            key="Drinks"
            value="Drinks"
          >
            Drinks
          </option>
          <option
            className="text-capitalize"
            key="Salads"
            value="Salads"
          >
            Salads
          </option>
        </select>
      </li>
      <li
        className=" col-12 "
      >
        <input
          className="btn bg-success hovered"
          id="submit"
          type="submit"
          value="save"
        />
      </li>
      <li
        className=" col-12 "
      >
        <button
          className="btn btn-block btn-lg col-12 bg-warning hovered"
          onClick={[Function]}
        >
          Back
        </button>
      </li>
    </ul>
  </form>
</div>
`;
